src/Features/CSharp/Portable/SignatureHelp/WithElementSignatureHelpProvider.cs

```diff
@@ -0,0 +1,114 @@
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.SignatureHelp;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CSharp.SignatureHelp;
+
+[ExportSignatureHelpProvider("WithElementSignatureHelpProvider", LanguageNames.CSharp), Shared]
+[method: ImportingConstructor]
+[method: Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+internal sealed partial class WithElementSignatureHelpProvider() : AbstractCSharpSignatureHelpProvider
+{
+    public override bool IsTriggerCharacter(char ch)
+        => ch is '(' or ',';
+
+    public override bool IsRetriggerCharacter(char ch)
+        => ch == ')';
+
+    private async Task<WithElementSyntax?> TryGetWithElementAsync(
+        Document document,
+        int position,
+        SignatureHelpTriggerReason triggerReason,
+        CancellationToken cancellationToken)
+    {
+        var withElement = await CommonSignatureHelpUtilities.TryGetSyntaxAsync<WithElementSyntax>(
+            document, position, triggerReason, IsTriggerToken, IsArgumentListToken, cancellationToken).ConfigureAwait(false);
+
+        return withElement?.ArgumentList is null ? null : withElement;
+    }
+
+    private bool IsTriggerToken(SyntaxToken token)
+        => SignatureHelpUtilities.IsTriggerParenOrComma<WithElementSyntax>(token, IsTriggerCharacter);
+
+    private static bool IsArgumentListToken(WithElementSyntax expression, SyntaxToken token)
+    {
+        return expression.ArgumentList != null &&
+            expression.ArgumentList.Span.Contains(token.SpanStart) &&
+            token != expression.ArgumentList.CloseParenToken;
+    }
+
+    protected override async Task<SignatureHelpItems?> GetItemsWorkerAsync(Document document, int position, SignatureHelpTriggerInfo triggerInfo, MemberDisplayOptions options, CancellationToken cancellationToken)
+    {
+        var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+        var withElement = await TryGetWithElementAsync(
+            document, position, triggerInfo.TriggerReason, cancellationToken).ConfigureAwait(false);
+        if (withElement?.Parent is not CollectionExpressionSyntax collectionExpression)
+            return null;
+
+        var semanticModel = await document.ReuseExistingSpeculativeModelAsync(withElement, cancellationToken).ConfigureAwait(false);
+        if (semanticModel.GetTypeInfo(collectionExpression, cancellationToken).ConvertedType is not INamedTypeSymbol collectionExpressionType)
+            return null;
+
+        var within = semanticModel.GetEnclosingNamedType(position, cancellationToken);
+        if (within == null)
+            return null;
+
+        var creationMethods = withElement
+            .GetCreationMethods(semanticModel, cancellationToken)
+            .WhereAsArray(s => s.IsEditorBrowsable(options.HideAdvancedMembers, semanticModel.Compilation))
+            .Sort(semanticModel, withElement.SpanStart);
```

Comment:
>Sort [](http://example.com/codeflow?start=13&length=4)

a bit of an aside, I had to look at where this sort is defined and it uses a ConcurrentDictionary internally for the sorting. Does that seem necessary to you?

Reply:
probably not :).  but not going to change that here.